src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LineCrossingWaterBodyCheck.java

```diff
@@ -412,12 +425,18 @@ private boolean canCrossWaterBody(final AtlasItem crossingItem,
                 final Set<Tuple<PolyLine, Set<Location>>> interactionsPerWaterbodyComponent = this
                         .getInteractionsPerWaterbodyComponent(waterbody, object,
                                 lineItem.asPolyLine(), flag);
-                final AtomicBoolean validCrossWaterbody = new AtomicBoolean(true);
+                this.validCrossWaterbody.getAndSet(true);
                 // Just need to see if the intersection points are allowed in OSM; if not flag them
                 if (interactionsPerWaterbodyComponent.isEmpty())
                 {
                     return false;
                 }
+                // refer to: https://github.com/osmlab/atlas-checks/issues/561
+                else if (interactionsPerWaterbodyComponent.size() == 1
+                        && this.lineItemsOffendingCrossingOnly.test(lineItem))
+                {
+                    this.handlingFootPath(interactionsPerWaterbodyComponent, lineItem);
```

Comment:
This seems redundant at the moment, as handlingFootPath only ever sets validCrossWaterbody to be true or does nothing and I believe validCrossWaterbody is always already true at this point.

Reply:
ones this condition met the logic is not following to else statement where the most of the cases processing. the confusion because of inverted return: !this.validCrossWaterbody.get();

Reply:
I get that the the final return inverted, and that if this else if is met then the else is bypassed. My issue is that when this else if is true then handlingFootPath is called, and handlingFootPath can only set validCrossWaterbody to true or do nothing, but validCrossWaterbody will always be true at the point handlingFootPath is called from this else if body.